backend/api/newsletters/forms.py

```diff
@@ -1,16 +1,25 @@
+import logging
+
 from django import forms
+
+from integrations.mailchimp import subscribe
 from newsletters.models import Subscription
 from strawberry_forms.forms import FormWithContext
 
+logger = logging.getLogger(__name__)
 
-class SubscribeToNewsletterForm(FormWithContext):
+
+class SussbscribeToNewsletterForm(FormWithContext):
     email = forms.EmailField()
 
     def save(self):
         email = self.cleaned_data.get("email")
-        subscription, _ = Subscription.objects.get_or_create(email=email)
 
-        return subscription
+        try:
+            return subscribe(email)
+        except Exception as e:
+            logger.error(e, exc_info=True)
```

Comment:
also `except Exception as e` feels very generic 🤔 

```suggestion
            logger.error("Unable to subscribe the user to mailchimp due to an error", e, exc_info=True)
```